### ISSUE TITLE
Search: Tab order mismatch in search modal (mobile)

### ISSUE THREAD

**Issue by katiebethbrown** [Automattic/jetpack#23367]
### Impacted plugin

Jetpack

### Steps to Reproduce

In the mobile view, the flex order of links in the search results is altered to place the thumbnail first. Because there are three links in each result, this results in a tab order mismatch between the visual sequence and the DOM order.

The visual order is thumbnail > title > breadcrumb title; the tab order is title > breadcrumb title > thumbnail.

The visual order must match the tab order.

Ideally, the thumbnail should be wrapped as a block-level link along with the post title, with an empty alt attribute. With that, the visible order does not matter, as it's a single object regardless. This would also help reduce the duplication of links and tedium of navigating the search results using a keyboard.

### Notes
_This issue was surfaced to team 51 by an a11y contractor working on a partner site._
